pkg/host/internal/infiniband/ib_guid_config_test.go

```diff
@@ -0,0 +1,277 @@
+package infiniband
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/vishvananda/netlink"
+
+	netlinkLibPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink"
+	netlinkMockPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink/mock"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/network"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/types"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/fakefilesystem"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/helpers"
+)
+
+var _ = Describe("IbGuidConfig", func() {
+	Describe("readJSONConfig", func() {
+		var (
+			tempDir    string
+			configPath string
+		)
+
+		BeforeEach(func() {
+			var err error
+			tempDir, err = os.MkdirTemp("", "ibguidconfig")
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			os.RemoveAll(tempDir)
+		})
+		It("should correctly decode a JSON configuration file with all fields present", func() {
+			mockJsonConfig := `[{"pciAddress":"0000:00:00.0","pfGuid":"00:00:00:00:00:00:00:00","guids":["00:01:02:03:04:05:06:07", "00:01:02:03:04:05:06:08"],"rangeStart":"00:01:02:03:04:05:06:08","rangeEnd":"00:01:02:03:04:05:06:FF"}]`
+
+			configPath = filepath.Join(tempDir, "config.json")
+			err := os.WriteFile(configPath, []byte(mockJsonConfig), 0644)
```

Comment:
do you want to use fakefilesystem instead? It is possible to create file with required content?

Reply:
Yep, employed the fakefs here